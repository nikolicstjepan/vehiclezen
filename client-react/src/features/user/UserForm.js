import React, { useState } from "react";

import InputGroup from "../../components/InputGroup";

export default function UserForm({ handleSubmit, actionText }) {
  const [form, setForm] = useState({ username: "", password: "" });

  const onSubmit = (e) => {
    e.preventDefault();
    handleSubmit(form);
  };

  const handleFormChange = ({ target }) => {
    const { name, value } = target;

    setForm({ ...form, [name]: value });
  };
  return (
    <div className="row justify-content-center">
      <div className="col-md-6">
        <form onSubmit={onSubmit}>
          <InputGroup name="username" value={form.username} onChange={handleFormChange} />
          <InputGroup
            name="password"
            type="password"
            value={form.password}
            onChange={handleFormChange}
          />
          <button type="submit" className="btn btn-primary">
            {actionText}
          </button>
        </form>
      </div>
    </div>
  );
}
