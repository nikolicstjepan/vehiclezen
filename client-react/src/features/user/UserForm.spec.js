import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";

import UserForm from "./UserForm";

describe("UserForm", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<UserForm actionText="test" handleSubmit={() => {}} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders inputGroup with username and password inputs", () => {
    const component = shallow(<UserForm actionText="test" handleSubmit={() => {}} />);
    expect(component.find("InputGroup")).toHaveLength(2);
    expect(component.find("InputGroup").at(0).prop("name")).toBe("username");
    expect(component.find("InputGroup").at(1).prop("name")).toBe("password");
  });

  test("requires username and password to be required", () => {
    const handleSubmit = jest.fn(() => {});
    const component = mount(<UserForm actionText="test" handleSubmit={handleSubmit} />);

    expect(component.find("InputGroup").at(0).find("input").prop("required")).toBeTruthy();
    expect(component.find("InputGroup").at(1).find("input").prop("required")).toBeTruthy();
  });

  test("renders button with given actionText", () => {
    const actionText = "test";
    const component = shallow(<UserForm actionText={actionText} handleSubmit={() => {}} />);

    expect(component.find("button")).toHaveLength(1);
    expect(component.find("button").text()).toBe(actionText);
  });

  test("calls handleSubmit when button is clicked", () => {
    const handleSubmit = jest.fn(() => {});
    const component = mount(<UserForm actionText="test" handleSubmit={handleSubmit} />);

    component.find("button").simulate("submit");
    expect(handleSubmit).toHaveBeenCalled();
  });
});
