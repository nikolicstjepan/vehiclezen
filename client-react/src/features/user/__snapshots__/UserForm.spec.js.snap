// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm snapshot renders 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-md-6"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          name="username"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <button
        className="btn btn-primary"
        type="submit"
      >
        test
      </button>
    </form>
  </div>
</div>
`;
